astropy_helpers/commands/build_ext.py

```diff
@@ -1,329 +1,180 @@
 import errno
 import os
-import re
-import shlex
 import shutil
-import subprocess
-import sys
-import textwrap
 
-from distutils import log, ccompiler, sysconfig
 from distutils.core import Extension
 from distutils.ccompiler import get_default_compiler
-from importlib import invalidate_caches
 from setuptools.command.build_ext import build_ext as SetuptoolsBuildExt
-from setuptools.command import build_py
 
-from ..utils import get_numpy_include_path, classproperty
-from ..version_helpers import get_pkg_version_module
+from ..distutils_helpers import get_main_package_directory
+from ..utils import get_numpy_include_path, import_file
 
 
-def should_build_with_cython(package, release=None):
-    """Returns the previously used Cython version (or 'unknown' if not
+def should_build_with_cython(previous_cython_version, is_release):
+    """
+    Returns the previously used Cython version (or 'unknown' if not
     previously built) if Cython should be used to build extension modules from
-    pyx files.  If the ``release`` parameter is not specified an attempt is
-    made to determine the release flag from `astropy.version`.
+    pyx files
     """
 
-    try:
-        version_module = __import__(package + '.cython_version',
-                                    fromlist=['release', 'cython_version'])
-    except ImportError:
-        version_module = None
-
-    if release is None and version_module is not None:
-        try:
-            release = version_module.release
-        except AttributeError:
-            pass
-
-    try:
-        cython_version = version_module.cython_version
-    except AttributeError:
-        cython_version = 'unknown'
-
     # Only build with Cython if, of course, Cython is installed, we're in a
     # development version (i.e. not release) or the Cython-generated source
     # files haven't been created yet (cython_version == 'unknown'). The latter
     # case can happen even when release is True if checking out a release tag
     # from the repository
     have_cython = False
     try:
-        import Cython  # noqa
+        from Cython import __version__ as cython_version  # noqa
         have_cython = True
     except ImportError:
         pass
 
-    if have_cython and (not release or cython_version == 'unknown'):
+    if have_cython and (not is_release or previous_cython_version == 'unknown'):
         return cython_version
     else:
         return False
 
 
-_compiler_versions = {}
-
-
-# TODO: I think this can be reworked without having to create the class
-# programmatically.
-def generate_build_ext_command(packagename, release):
+class AstropyBuildExt(SetuptoolsBuildExt):
```

Comment:
It might confuse people that this is the "astropy builder" given that it's really not astropy itself...

```suggestion
class HelpersBuildExt(SetuptoolsBuildExt):
```
(of course accepting this would also require changing the names below as well...)

Reply:
I was mostly going with the naming convention used for other builders, e.g. ``AstropyBuildDocs`` but I see your point. I might change it to ``AstropyHelpersBuildExt`` instead of ``HelpersBuildExt`` though?